GenericTest.py

```diff
@@ -286,8 +287,35 @@ def basics(self):
                         if result is not None:
                             results.append(result)
 
+            # Perform an automatic check for an error condition
+            results.append(self.check_404_path(api))
+
         return results
 
+    def check_404_path(self, api_name):
+        api = self.apis[api_name]
+        error_code = 404
+        invalid_path = str(uuid.uuid4())
+        url = "{}/{}".format(api["url"].rstrip("/"), invalid_path)
+        test = Test("GET /x-nmos/{}/{}/{} ({})".format(api_name, api["version"], invalid_path, error_code),
+                    self.auto_test_name())
+
+        status, response = self.do_request("GET", url)
```

Comment:
Very minor gripe. Everywhere else (I think!) the first return of `do_request` is called something like `valid`. I think `status` is a little confusing here as we're testing at the status code. I know there's another `valid` down below though...

Reply:
Looks like I copied it from the one other place status was used. I'll fix both.

Reply:
Ah! Ta.